models/review.go

```diff
@@ -442,3 +446,55 @@ func InsertReviews(reviews []*Review) error {
 
 	return sess.Commit()
 }
+
+// SwithchRewiewRequest switch Rewiew Request status
+func SwithchRewiewRequest(issue *Issue, reviewer *User, doer *User) (isRequest bool, comment *Comment, err error) {
+	reviews, err := GetReviewersByIssueID(issue.ID)
+	if err != nil {
+		return
+	}
+	var get *Review
+	for _, rev := range reviews {
+		if rev.ReviewerID == reviewer.ID {
+			get = rev
+			break
+		}
+	}
+
+	if get != nil && get.Type == ReviewTypeRequest {
+		_, err = x.Delete(get)
+		if err != nil {
+			return
+		}
+		isRequest = false
+	} else {
+		_, err = CreateReview(CreateReviewOptions{
+			Type:     ReviewTypeRequest,
+			Issue:    issue,
+			Reviewer: reviewer,
+			Content:  doer.Name,
+			Official: false,
```

Comment:
The review request should replace any official reviews for that reviewer if the reviewer is "official".

Reply:
sorry , I have ignored the role of this member variable before, is it used to judge whether the review is valid or not? If so, I think it should be determined by the poster of the request rather than reviewer who is requested , and when Reject can block Merge, ReviewRequest should also hve the same function. Is my think right?

Reply:
Hello, I have modified it according to my understanding, but I am not sure whether it is correct.